driver/modern_bpf/programs/attached/events/security_file_mprotect.bpf.c

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-2.0-only OR MIT
+/*
+ * Copyright (C) 2023 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+/*=============================== ENTER EVENT ===========================*/
+
+#include <helpers/interfaces/variable_size_event.h>
+#include <helpers/extract/extract_from_kernel.h>
+
+SEC("lsm/file_mprotect")
+int BPF_PROG(file_mprotect,
+	     struct vm_area_struct *vma,
+	     unsigned long reqprot,
+	     unsigned long prot,
+	     int ret)
+{
+        struct auxiliary_map *auxmap = auxmap__get();
+        if(!auxmap)
+        {
+                return 0;
+        }
+
+        auxmap__preload_event_header(auxmap, PPME_LSM_SECURITY_FILE_MPROTECT_E);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: vm_start (PT_UINT64) */
+	unsigned long vm_start = extract__vm_start(vma);
+	auxmap__store_u64_param(auxmap, vm_start);
+
+	/* Parameter 2: vm_end (PT_UINT64) */
+	unsigned long vm_end = extract__vm_end(vma);
+	auxmap__store_u64_param(auxmap, vm_end);
+
+	/* Parameter 3: reqprot (type: PT_FLAGS32) */
+	auxmap__store_u32_param(auxmap, reqprot);
+
+	/* Parameter 4: prot (type: PT_FLAGS32)*/
+	auxmap__store_u32_param(auxmap, prot);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+	//auxmap__submit_event(auxmap, ctx);
```

Comment:
The auxmap API is wrong (we should be using fixed-size ringbuf), but to illustrate the problem I'm facing, let's just go with this for now.

If this line is not commented out, the verifier fails:

```
libbpf: prog 'file_mprotect': BPF program load failed: Invalid argument
libbpf: prog 'file_mprotect': -- BEGIN PROG LOAD LOG --
processed 623 insns (limit 1000000) max_states_per_insn 4 total_states 50 peak_states 48 mark_read 3
-- END PROG LOAD LOG --
libbpf: prog 'file_mprotect': failed to load: -22
libbpf: failed to load object 'bpf_probe'
libbpf: failed to load BPF skeleton 'bpf_probe': -22
libpman: failed to load BPF object (errno: 22 | message: Invalid argument)/home/dave/src/libs/test/libscap/test_suites/engines/modern_bpf/modern_bpf.cpp:148: Failure
Value of: !h || ret != 0
  Actual: true
Expected: false
```

Rather than digging deeper into this now, could I get a sanity check to see if this is the correct approach generally given the guidance in #963? @FedeDP

Reply:
I agree with this approach; i don't know why this is failing though ;)
Anyway, as shared privately, we surely need more work to understand how to integrate LSM/kprobes inside current libs architecture:
* we will have a single event (instead of pushing a fake empty exit event), while we always had enter/exit events until now
* kprobes/LSM don't carry a return code, therefore we need to understand how to deal with this too
* how to avoid enriching same state from different hooks (ie: both kprobe and syscalls)?

In the end, there are multiple questions looking for a good answer before we can start.
Hopefully maintainers will share some proposals soonish :)